### PR TITLE
Fix/failing test suites

### PR DESCRIPTION
**Pull request by Roman-batut**
This pull request includes several changes to improve the test coverage and functionality of the ViewModel tests. The most important changes involve adding mock route parameters, updating the `useCameraViewModel` and `useMapScreenViewModel` hooks, and enhancing the `createChallenge` function.

### Mock route parameters:

* Added `mockRoute` with `group_id` parameter to the `cameraViewModel.test.tsx` file and updated all `useCameraViewModel` calls to include `mockRoute`. [[1]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dR26-R31) [[2]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL37-R43) [[3]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL50-R56) [[4]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL68-R74) [[5]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL97-R103) [[6]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dR118) [[7]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL124-R131)

### `useCameraViewModel` updates:

* Updated `useCameraViewModel` calls in `cameraViewModel.test.tsx` to include `mockRoute` to ensure the tests cover route parameter usage. [[1]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL37-R43) [[2]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL50-R56) [[3]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL68-R74) [[4]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL97-R103) [[5]](diffhunk://#diff-146c6414635d84fa6cabafb9132c0c7a18c7c99d2a0f9a32e615361d138a148dL124-R131)

### `useMapScreenViewModel` updates:

* Replaced the custom `MockGeoPoint` with the `GeoPoint` class from `firebase/firestore` in `mapScreenViewModel.test.tsx`. Updated all `useMapScreenViewModel` calls to include `undefined` for the third parameter. [[1]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5R10) [[2]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L26-R39) [[3]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L75-R71) [[4]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L100-R94) [[5]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L118-R117) [[6]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L137-R131) [[7]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L153-R147) [[8]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L173-R167) [[9]](diffhunk://#diff-7f5ec2b7a4c1cf6ca370c7355372d133488c04235ee2932d47beb31774d849d5L187-R181)

### `createChallenge` function enhancements:

* Added `group_id` parameter to the `createChallenge` function calls in `createChallengeViewModel.test.tsx` and `challengeBuilder.test.tsx`. [[1]](diffhunk://#diff-43c9d25de18fe518332acaed76bcd130fcfb8eb4ee570c13305f932a3a4553e3R155-R159) [[2]](diffhunk://#diff-2185724de776e546315ad90954fd85c289d493bb451eddcc61d89807e1208238R68) [[3]](diffhunk://#diff-2185724de776e546315ad90954fd85c289d493bb451eddcc61d89807e1208238L95-R96)

### General improvements:

* Mocked `FirestoreCtrl` in `createChallengeViewModel.test.tsx` and updated related tests to use the mock implementation. [[1]](diffhunk://#diff-43c9d25de18fe518332acaed76bcd130fcfb8eb4ee570c13305f932a3a4553e3R8) [[2]](diffhunk://#diff-43c9d25de18fe518332acaed76bcd130fcfb8eb4ee570c13305f932a3a4553e3R23-L33)

### CODE REVIEW

File: __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx [hunk 1]
```diff
@@ -97,12 +81,17 @@ describe("useMapScreenViewModel", () => {
     });
 
     const { result } = renderHook(() =>
-      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation),
+      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation, undefined),
```
Comment (margdsq) [inline __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx:84]: This is not very clear, could your precise it's for the groups that are not used in the tests for now ?

File: __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx [hunk 2]
```diff
@@ -115,12 +104,17 @@ describe("useMapScreenViewModel", () => {
     );
 
     const { result } = renderHook(() =>
-      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation),
+      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation, undefined),
```
Comment (margdsq) [inline __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx:107]: Same here.

File: __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx [hunk 3]
```diff
@@ -150,7 +144,7 @@ describe("useMapScreenViewModel", () => {
...
```
Comment (margdsq) [inline __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx:147]: Same here.

File: __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx [hunk 4]
```diff
@@ -170,7 +164,7 @@ describe("useMapScreenViewModel", () => {
...
```
Comment (margdsq) [inline __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx:167]: Same here.

File: __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx [hunk 5]
```diff
@@ -184,7 +178,7 @@ describe("useMapScreenViewModel", () => {
     jest.spyOn(console, "error").mockImplementationOnce(() => {});
 
     const { result } = renderHook(() =>
-      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation),
+      useMapScreenViewModel(mockFirestoreCtrl, mockNavigation, undefined),
```
Comment (margdsq) [inline __tests__/ViewModel-tests/map/mapScreenViewModel.test.tsx:181]: Same here.

File: __tests__/types/challengeBuilder.test.tsx [hunk 6]
```diff
@@ -92,7 +93,7 @@ describe("createChallenge", () => {
       .mockRejectedValue(new Error("Firestore error"));
 
     await expect(
-      createChallenge(mockFirestoreCtrl, "Test", "Description", null),
```
Comment (margdsq) [inline __tests__/types/challengeBuilder.test.tsx:95]: Same here, what's null ?

**Review by margdsq — CHANGES_REQUESTED** [review]
Thank you for this much necessary PR !

File: __tests__/types/challengeBuilder.test.tsx (hunk 6 above)
Comment (Roman-batut) [reply __tests__/types/challengeBuilder.test.tsx:95]: null here is the location, as this test doesn't use it and mocking the location would require some effort where we don't really need it here.

**Review by margdsq — APPROVED** [review]
LGTM !

**Pull request merged**